chia/data_layer/data_layer.py

```diff
@@ -359,25 +305,48 @@ async def periodically_fetch_data(self) -> None:
                     break
                 except aiohttp.client_exceptions.ClientConnectorError:
                     pass
+                except asyncio.CancelledError:
+                    return
```

Comment:
this is the main task (so nothing awaits it, except for the DataLayer object). I've tried to return in cases like this and raise otherwise (where something else awaits the function).

Reply:
I think the thing calling this should manage the cancellation, if even that.  Otherwise any time we reuse code, maybe for tests, we have to deal with the fact it doesn't cancel properly.  Code should not cancel termination unless it initiated the cancellation of sub-code.  Asyncio doesn't offer a means by which to differentiate this accurately at this point so at best it is pretty hard to do.

Reply:
ok, I've raised it inside this task as well, and caught it only when doing `self.periodically_manage_data_task.cancel()`